source/AI.cpp

```diff
@@ -1087,7 +1087,7 @@ bool AI::HasHelper(const Ship &ship, const bool needsFuel)
 
 
 // Pick a new target for the given ship.
-shared_ptr<Ship> AI::FindTarget(const Ship &ship) const
+shared_ptr<Ship> AI::FindTarget(Ship &ship) const
```

Comment:
Since this function now modifies `ship`, making it return the parent is a bit redundant. So I propose to (at the end of this function) you can set the new parent for the ship.